web/pgadmin/browser/server_groups/servers/__init__.py

```diff
@@ -714,6 +718,19 @@ def delete(self, gid, sid):
                     server_name = s.name
                     get_driver(PG_DEFAULT_DRIVER).delete_manager(s.id)
                     db.session.delete(s)
+                    if not config.SERVER_MODE \
+                            or not config.DISABLED_LOCAL_PASSWORD_STORAGE:
```

Comment:
You can simply set `DISABLED_LOCAL_PASSWORD_STORAGE=True` for server mode so that you don't have to put all the `or` conditions below and make it more readable.